plugins/fossa/src/components/useProjectName.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Entity } from '@backstage/catalog-model';
+
+export const FOSSA_PROJECT_NAME_ANNOTATION = 'fossa.io/project-name';
+
+export const useProjectName = (entity: Entity) => {
+  return entity?.metadata.annotations?.[FOSSA_PROJECT_NAME_ANNOTATION] ?? '';
```

Comment:
Maybe this is better to do `?? undefined` instead of empty string. Just for semantics really?

Reply:
I updated it to return `undefined` when no value is set.

It seems to be [a](https://github.com/backstage/backstage/blob/c4660094cf33c5c5d9b4c7c1d09e72d4db81dbd8/plugins/cloudbuild/src/components/useProjectName.ts#L24) [pattern](https://github.com/backstage/backstage/blob/c4660094cf33c5c5d9b4c7c1d09e72d4db81dbd8/plugins/github-actions/src/components/useProjectName.ts#L24) [in](https://github.com/backstage/backstage/blob/c4660094cf33c5c5d9b4c7c1d09e72d4db81dbd8/plugins/sentry/src/components/useProjectSlug.ts#L22) [other](https://github.com/backstage/backstage/blob/c4660094cf33c5c5d9b4c7c1d09e72d4db81dbd8/plugins/jenkins/src/components/useProjectSlugFromEntity.ts#L23) [plugins](https://github.com/backstage/backstage/blob/c4660094cf33c5c5d9b4c7c1d09e72d4db81dbd8/plugins/sonarqube/src/components/useProjectKey.ts#L22), [too](https://github.com/backstage/backstage/blob/c4660094cf33c5c5d9b4c7c1d09e72d4db81dbd8/plugins/circleci/src/state/useBuilds.ts#L108). Maybe worth taking a look to that later.